azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,958 @@
+			},
+			"notification": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"email": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"send_to_subscription_administrator": {
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
+									},
+									"send_to_subscription_co_administrator": {
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
+									},
+									"custom_emails": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+							Set: resourceAzureRmAutoscaleDefaultHash,
+						},
+						"webhook": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							MinItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"service_uri": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"properties": {
+										Type:     schema.TypeMap,
+										Optional: true,
+									},
+								},
+							},
+							Set: resourceAzureRmAutoscaleDefaultHash,
+						},
+					},
+				},
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"target_resource_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmAutoscaleSettingCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	asClient := meta.(*ArmClient).autoscaleSettingsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroupName := d.Get("resource_group_name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	enabled := d.Get("enabled").(bool)
+	targetResourceURI := d.Get("target_resource_id").(string)
+	tags := d.Get("tags").(map[string]interface{})
+	expandedTags := expandTags(tags)
+
+	profiles, err := expandAzureRmAutoscaleProfile(d)
+	if err != nil {
+		return fmt.Errorf("Error parsing Autoscale Profile: %+v", err)
+	}
+
+	notifications, err := expandAzureRmAutoscaleNotification(d)
+	if err != nil {
+		return err
+	}
+
+	parameters := insights.AutoscaleSettingResource{
+		Name:     &name,
+		Location: &location,
+		Tags:     expandedTags,
+		AutoscaleSetting: &insights.AutoscaleSetting{
+			Name:              &name,
+			Enabled:           &enabled,
+			TargetResourceURI: &targetResourceURI,
+			Profiles:          &profiles,
+			Notifications:     &notifications,
+		},
+	}
+
+	_, err = asClient.CreateOrUpdate(ctx, resourceGroupName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	read, err := asClient.Get(ctx, resourceGroupName, name)
+	if err != nil {
+		return err
```

Comment:
*minor* It would be better to be consistent with other error message (for example, including resource group name in the message).